src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Replace.cs

```diff
@@ -193,11 +193,11 @@ private static string Replace(MatchEvaluator evaluator, Regex regex, string inpu
             {
                 state.prevat = input.Length;
 
-                regex.Run(input, startat, ref state, (ref (SegmentStringBuilder segments, MatchEvaluator evaluator, int prevat, string input, int count) state, Match match) =>
+                regex.Run(input, startat, ref state, static (ref (SegmentStringBuilder segments, MatchEvaluator evaluator, int prevat, string input, int count) state, Match match) =>
                 {
                     state.segments.Add(state.input.AsMemory(match.Index + match.Length, state.prevat - match.Index - match.Length));
                     state.prevat = match.Index;
-                    state.segments.Add(evaluator(match).AsMemory());
+                    state.segments.Add(state.evaluator(match).AsMemory());
```

Comment:
This is the fix.

Reply:
Just curious about how you found this?

Reply:
Wednesdays are a busy meeting day for me, and I find my mind is able to stay more focused on them if I give it little mindless side projects, like walking in ILSpy through all "DisplayClass"es in all of netcoreapp looking for ones that don't belong :)

Reply:
Next task is to look a dumps all day.